File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs.size();
        if(n==0)return "";

        sort(strs.begin(),strs.end());

        int size1 = strs[0].size();
        int size2 = strs[n-1].size();

        int i=0;
        string answer = "";
        while(i < min(size1, size2))
        {
            if(strs[0][i] == strs[n-1][i])
            {
                answer+= strs[0][i];
            }
            else break;

            i++;
        }
        return answer;
        
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;

        Arrays.sort(strs);

        int i = 0;
        int j  = 0;

        StringBuilder answer = new StringBuilder();

        while( i < strs[0].length() && j < strs[n-1].length())
        {
            char ch1 = strs[0].charAt(i);
            char ch2 = strs[n-1].charAt(j);
            i++;
            j++;

            if(ch1 == ch2)
            {
                answer.append(ch1);
            }
            else
            {
                return answer.toString();
            }
        }
        return answer.toString();
        
    }
}